### PR TITLE
tests/provisioning: restrict feed access and empower test wrapper scripts

### PR DESCRIPTION
**Pull request by amstewart**
This patchset does two primary things:
1. It removes the `base-feed.conf` opkg sources prior to installing the newly build grub-gateway and provisioning test scripts. This keeps the tests fixtures from accidently downloading old (already published) feed content, if the locally-built feeds don't have what is necessary. We would prefer for the test to just fail in that case, rather than falsely passing.
2. It adds a wrapper script for running the migration tests, and empowers it - and the extant `test_provisioning.sh` script - to pass the lower-leverl tcl-expect script the files they need, and to parse stdout for the test results. The goal here is to (a) allow external builders to easily call these same wrapper scripts and (b) to generally move test logic out of the AZDO component. There will be an associated AZDO PR, which interfaces with the new wrapper scripts.

## Testing
Both test_ targets build on my machine.

@ni/rtos

### CODE REVIEW

**Comment by amstewart** [comment]
Some relevant parts of the test log, showing that the base feeds are now removed.
```
# test_provisioning.log
admin@NI-Unknown-525412a2fddb:~# rm -f /etc/opkg/base-feeds.conf
admin@NI-Unknown-525412a2fddb:~# touch /etc/opkg/testing.conf
admin@NI-Unknown-525412a2fddb:~# echo src test-feed.0 http://10.0.2.2:8990/feed.0^M0 [trusted=yes] >>/etc/opkg/testing.conf
admin@NI-Unknown-525412a2fddb:~# echo src test-feed.1 http://10.0.2.2:8990/feed.1^M1 [trusted=yes] >>/etc/opkg/testing.conf
admin@NI-Unknown-525412a2fddb:~# echo src test-feed.2 http://10.0.2.2:8990/feed.2^M2 [trusted=yes] >>/etc/opkg/testing.conf
admin@NI-Unknown-525412a2fddb:~# cat /etc/opkg/testing.conf
src test-feed.0 http://10.0.2.2:8990/feed.0 [trusted=yes]
src test-feed.1 http://10.0.2.2:8990/feed.1 [trusted=yes]
src test-feed.2 http://10.0.2.2:8990/feed.2 [trusted=yes]
admin@NI-Unknown-525412a2fddb:~# opkg update
Downloading http://nickdanger.amer.corp.natinst.com/feeds/dist/Packages.gz.
Downloading http://nickdanger.amer.corp.natinst.com/feeds/dist/Packages.asc.
Updated source 'NI-dist'.
Downloading http://10.0.2.2:8990/feed.0/Packages.
Updated source 'test-feed.0'.
Downloading http://10.0.2.2:8990/feed.1/Packages.
Updated source 'test-feed.1'.
Downloading http://10.0.2.2:8990/feed.2/Packages.
Updated source 'test-feed.2'.
admin@NI-Unknown-525412a2fddb:~# opkg install init-restore-mode-ptest
Downloading http://10.0.2.2:8990/feed.2/init-restore-mode-lic_1.0-r0.1649_x64.ipk.
Downloading http://10.0.2.2:8990/feed.0/ptest-runner-lic_2.1+git1+41b7f4814d-r0.255_core2-64.ipk.
Downloading http://10.0.2.2:8990/feed.0/ptest-runner_2.1+git1+41b7f4814d-r0.255_core2-64.ipk.
Downloading http://10.0.2.2:8990/feed.2/init-restore-mode-ptest_1.0-r0.1649_x64.ipk.
Installing init-restore-mode-lic (1.0) on root
Installing ptest-runner-lic (2.1+git1+41b7f4814d) on root
Installing ptest-runner (2.1+git1+41b7f4814d) on root
Installing init-restore-mode-ptest (1.0) on root
```

The migration test is similar.

File: scripts/tests/provisioning/test_rauc_provision.expect [hunk 1]
```diff
@@ -49,6 +49,7 @@ set timeout 180
 # configure opkg
 expect {
 	-re "$RE_PROMPT" {
+		send "rm -f /etc/opkg/base-feeds.conf\r"
```
Comment (shruthi-ravi) [inline scripts/tests/provisioning/test_rauc_provision.expect]: Quick comment while I remember, how about removing dist.conf too?

Comment (amstewart) [reply scripts/tests/provisioning/test_rauc_provision.expect]: Darn; that completely slipped my mind. I'll see if I can remove that as well.

**Comment by amstewart** [comment]
## Version 2
* Also remove the `/etc/opkg/NI-dist.conf` feed source for the same reason. [commit](https://github.com/amstewart/nilrt/commit/271f22e0e747311deacc1bfb87a5e56f6087af68).

### Testing
* Rebuilt test_ targets on my dev machine and confirmed that they still work and look good.

**Comment by amstewart** [comment]
I'm not sure why github isn't updating this PR, but there is a 5th commit on the dev branch. :/

**Comment by amstewart** [comment]
TIL: if you force-push the same git ref as the merge target to the merge branch, GH will automatically close the PR and say that it has been merged. :/

I have created a [new PR](https://github.com/ni/nilrt/pull/61), which has *all* of the commits that are supposed to be in this patchset. Hopefully that one won't be broken.

**Pull request closed without merging**